bloptools/bayesian/objectives.py

```diff
@@ -0,0 +1,139 @@
+from collections.abc import Sequence
+from dataclasses import dataclass
+from typing import Tuple, Union
+
+import numpy as np
+import pandas as pd
+
+numeric = Union[float, int]
+
+DEFAULT_MINIMUM_SNR = 1e2
+OBJ_FIELDS = ["description", "target", "active", "limits", "weight", "log", "n", "snr", "min_snr"]
+
+
+class DuplicateNameError(ValueError):
+    ...
+
+
+def _validate_objectives(objectives):
+    names = [obj.name for obj in objectives]
+    unique_names, counts = np.unique(names, return_counts=True)
+    duplicate_names = unique_names[counts > 1]
+    if len(duplicate_names) > 0:
+        raise DuplicateNameError(f'Duplicate name(s) in supplied objectives: "{duplicate_names}"')
+
+
+@dataclass
+class Objective:
+    name: str
+    description: str = None
+    target: Union[float, str] = "max"
+    log: bool = False
+    weight: numeric = 1.0
+    active: bool = True
+    limits: Tuple[numeric, numeric] = None
+    min_snr: numeric = DEFAULT_MINIMUM_SNR
+    units: str = None
+
+    def __post_init__(self):
+        if self.limits is None:
+            if self.log:
+                self.limits = (0, np.inf)
+            else:
+                self.limits = (-np.inf, np.inf)
+
+        if type(self.target) is str:
+            if self.target not in ["min", "max"]:
+                raise ValueError("'target' must be either 'min', 'max', or a number.")
+
+        # self.device = Signal(name=self.name)
+
+    @property
+    def label(self):
+        return f"{'log ' if self.log else ''}{self.description}"
+
+    @property
+    def summary(self):
+        series = pd.Series()
+        for col in OBJ_FIELDS:
+            series[col] = getattr(self, col)
+        return series
+
+    def __repr__(self):
+        return self.summary.__repr__()
+
+    @property
+    def noise(self):
+        return self.model.likelihood.noise.item() if hasattr(self, "model") else None
+
+    @property
+    def snr(self):
+        return np.round(1 / self.model.likelihood.noise.sqrt().item(), 1) if hasattr(self, "model") else None
+
+    @property
+    def n(self):
+        return self.model.train_targets.shape[0] if hasattr(self, "model") else 0
```

Comment:
Short docstrings for those properties will help.

Also, is `self.min_snr` (defaulting to `DEFAULT_MINIMUM_SNR`) used anywhere?

Reply:
Yes, it's used in the agent